Bio/PDB/Entity.py

```diff
@@ -183,6 +183,33 @@ def id(self, value):
         self._id = value
         self._reset_full_id()
 
+    def strictly_equals(self, other: "Entity") -> bool:
+        """Compare this entity to the other entity for equality.
+
+        Recursively compare the children of this entity to the other entity's children.
+        Compare most properties including names and atomic coordinates.
+
+        :param other: The entity to compare this entity with
+        :type other: Entity
+        :return: Whether the two entities are strictly equal
+        :rtype: bool
+        """
+        if not isinstance(other, type(self)):
+            return False
+
+        if self.id != other.id:
+            return False
+
+        if len(self.child_list) != len(other.child_list):
+            return False
+
+        for left_child, right_child in zip(self.child_list, other.child_list):
+            assert hasattr(left_child, "strictly_equals")
```

Comment:
Isn't this covered by the `isinstance` above?

Reply:
I think that in standard usage of the Entity class, you would never expect this assertion to fail. All children should be other entities or atoms, all of which have `strictly_equals`. But I don't think that the assertion is logically or technically guaranteed to be true, so I add the assertion statement to indicate that there is an issue in case it fails.

Reply:
Alright!